chunky/src/java/se/llbit/math/Intersectable.java

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2014-2021 Jesper Öqvist <jesper@llbit.se>
+ * Copyright (c) 2014-2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.math;
+
+/**
+ * Anything which can intersect a ray in space.
+ */
+public interface Intersectable {
```

Comment:
The only `Intersectable` thing is the BVH at the moment. Is this interface needed?

Reply:
I'd like to use it in upcoming changes/plugins soon. It could additionally be marked as `@FunctionalInterface`